metrics.go

```diff
@@ -269,10 +269,9 @@ func (m *Metrics) emitRuntimeStats() {
 	m.lastNumGC = num
 }
 
-// Inserts a string value at an index into the slice
-func insert(i int, v string, s []string) []string {
-	s = append(s, "")
-	copy(s[i+1:], s[i:])
-	s[i] = v
-	return s
+// Creates a new slice with the provided string value as the first element
+// and the provided slice values as the remaining values.
+// Ordering of the values in the provided input slice is kept in tact in the output slice.
+func insertAtBeginning(v string, s []string) []string {
+	return append([]string{v}, s...)
 }
```

Comment:
why not just leave it as a flexible insert, but correct the behavior:
```lang=go
func(i int, v string, s []string) []string {
	return append(s[:i], append([]string{v}, s[i:]...)...)
}

See https://github.com/golang/go/wiki/SliceTricks

Reply:
the thing is the flexible capability is not used; this is only called with index 0, and seems to have been as such for several years.
Unless there is a clear use case for the flexible version, we should keep it as simple as possible both for easier maintenance and for the slightly better performance (which is impactful for something like a metrics library)

@armon , was there a future use case for having the flexible version for any index?

if I'm reading this correctly, it is potentially insightful to note from @zmt's link that the current implementation is listed as a means to *avoid* creation of an additional slice when inserting at an arbitrary index, though neglects to mention the problem caused by that (the issue this PR fixes)